Source/Dafny/RefinementTransformer.cs

```diff
@@ -1351,7 +1351,18 @@ public RefinementTransformer(Program p)
 
       var invs = cOld.Invariants.ConvertAll(refinementCloner.CloneAttributedExpr);
       invs.AddRange(cNew.Invariants);
-      var r = new RefinedWhileStmt(cNew.Tok, cNew.EndTok, guard, invs, decr, refinementCloner.CloneSpecFrameExpr(cOld.Mod), MergeBlockStmt(cNew.Body, cOld.Body));
+      BlockStmt newBody;
+      if (cOld.Body == null && cNew.Body == null) {
```

Comment:
I have no idea. I'm guessing an arbitrary letter was once introduced so that the identifiers were not the C# keyword `new` but instead the identifier `cNew`.